backend/test/backend.go

```diff
@@ -5,10 +5,12 @@ import (
 
 	"github.com/cschleiden/go-workflows/backend"
 	"github.com/cschleiden/go-workflows/backend/history"
+	"github.com/cschleiden/go-workflows/diag"
 )
 
 type TestBackend interface {
 	backend.Backend
+	diag.Backend
```

Comment:
I'd prefer if we kept this separate and added a new set of tests cases for diag backends instead

Reply:
sure I can make a change for that.

Reply:
removed in favor of simply asserting the implementation type in the test